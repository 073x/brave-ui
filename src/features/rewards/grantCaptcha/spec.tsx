/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import GrantCaptcha from './index';

describe('Grant captcha tests', () => {
  const baseComponent = (props?: object) => <GrantCaptcha id='captcha' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#captcha').length
      expect(assertion).toBe(1)
    })
  })
})
