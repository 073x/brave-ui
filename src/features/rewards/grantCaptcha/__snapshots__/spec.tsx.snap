// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grant captcha tests basic tests matches the snapshot 1`] = `
.c0 {
  text-align: center;
  width: 100%;
  margin: 28px 0 0;
}

.c5 {
  width: 333px;
  height: 296px;
  margin-left: -32px;
  margin-top: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-flex-basis: 80px;
  -ms-flex-preferred-size: 80px;
  flex-basis: 80px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  -webkit-flex-basis: 130px;
  -ms-flex-preferred-size: 130px;
  flex-basis: 130px;
  font-family: Muli,sans-serif;
  font-size: 14px;
  line-height: 1.29;
  color: #686978;
  text-align: left;
  padding-left: 13px;
}

.c3 {
  width: 60px;
  height: 52px;
}

<div
  className="c0"
  id="captcha"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        className="c3"
        draggable={true}
        onDragStart={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="c4"
    >
      MISSING: dndCaptchaText1
       
      <b />
       
      MISSING: dndCaptchaText2
    </div>
  </div>
  <img
    className="c5"
    draggable={false}
    onDragOver={[Function]}
    onDrop={[Function]}
  />
</div>
`;
